### PR TITLE
More supported data types, docs update

### PR DESCRIPTION
**Pull request by matteofigus**
*Issue #, if available:*
We only support a number of data types. With this change, I updated the docs, plus I updated the frontend to make it easier to show what's allowed or not.

Note: Future PRs are coming for supporting more types.
I left out here particularly dates, decimal, and complex structures. One about struct in particular, will allow all the allowed types when nested inside structs. When that will happen, I'll update the docs too.

Screenshot 2020-07-03 at 15 52 16
Screenshot 2020-07-03 at 15 52 46

*PR Checklist:*

- [x] Unit tests (and integration tests if applicable) provided
- [x] All tests pass
- [x] Pre-commit checks pass
- [x] Debugging code removed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/src/utils/glueSerializer.js [hunk 1]
```diff
@@ -1,6 +1,89 @@
...
+    "double",
+    "float",
+    "int",
+    "smallint",
+    "string",
+    "tinyint",
+    "varchar"
+  ];
+
+  const columnMapper = c => {
```
Comment (HieronymusLex) [inline frontend/src/utils/glueSerializer.js:19]: I think we could probably do with a test for this? also it's quite hard to follow what's happening and some tests would help with that

File: frontend/src/utils/glueSerializer.js [hunk 2]
```diff
@@ -1,6 +1,89 @@
...
+        const isStruct = rest.startsWith(STRUCTPREFIX);
+        const isStructArray = rest.startsWith(ARRAYSTRUCTPREFIX);
+
+        if (isStruct || isStructArray) {
+          const prefix = isStruct ? STRUCTPREFIX : ARRAYSTRUCTPREFIX;
+          let nOpenedTags = isStruct ? 1 : 2;
+          let endIndex = -1;
+          const structSchema = [...rest.substr(prefix.length)];
+          structSchema.some((c, i) => {
+            if (c === "<") nOpenedTags++;
```
Comment (HieronymusLex) [inline frontend/src/utils/glueSerializer.js:60]: not too sure about callback passed to `some` mutating vars from the outer scope. doesn't seem like a good approach from an functional perspective (not that were doing FP in this project!!!)

File: frontend/src/utils/glueSerializer.js (hunk 1 above)
Comment (HieronymusLex) [reply frontend/src/utils/glueSerializer.js:19]: can we break this up a bit to make it more readable?

**Comment by HieronymusLex** [comment]
@matteofigus I think we should be updating the changelog as we go e.g. add a new section to the changelog for the next version (marked as draft/pending) then we add to it as we merge PRs - thoughts?

File: frontend/src/utils/glueSerializer.js (hunk 1 above)
Comment (matteofigus) [reply frontend/src/utils/glueSerializer.js:80]: done

Comment (matteofigus) [reply frontend/src/utils/glueSerializer.js:80]: done

File: frontend/src/utils/glueSerializer.js (hunk 2 above)
Comment (matteofigus) [reply frontend/src/utils/glueSerializer.js]: I think you're right. Moved to for, which makes it actually more readable

File: backend/lambdas/tasks/generate_queries.py [hunk 3]
```diff
@@ -114,9 +114,11 @@ def convert_to_col_type(val, col, table):
 
     col_type = column["Type"]
 
-    if col_type == "string" or col_type == "varchar":
+    if col_type in ["char", "string", "varchar"]:
```
Comment (ctd) [inline backend/lambdas/tasks/generate_queries.py:117]:
```suggestion
    if col_type in ("char", "string", "varchar"):
```

Nitpick. Tuples are more appropriate here.

File: backend/lambdas/tasks/generate_queries.py [hunk 4]
```diff
@@ -114,9 +114,11 @@ def convert_to_col_type(val, col, table):
...
         return str(val)
-    if col_type == "int" or col_type == "bigint":
+    if col_type in ["bigint", "int", "smallint", "tinyint"]:
```
Comment (ctd) [inline backend/lambdas/tasks/generate_queries.py:119]:
```suggestion
    if col_type in ("bigint", "int", "smallint", "tinyint"):
```

Nitpick. Tuples are more appropriate here.

File: backend/lambdas/tasks/generate_queries.py [hunk 5]
```diff
@@ -114,9 +114,11 @@ def convert_to_col_type(val, col, table):
...
         return int(val)
+    if col_type in ["double", "float"]:
```
Comment (ctd) [inline backend/lambdas/tasks/generate_queries.py:121]:
```suggestion
    if col_type in ("double", "float"):
```

Nitpick. Tuples are more appropriate here.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,13 @@
 # Change Log
 
+## 0.6 (unreleased)
+
+### Summary
+
+- [#173](https://github.com/awslabs/amazon-s3-find-and-forget/pull/173): Show
```
Comment (ctd) [inline CHANGELOG.md:7]: Since there's two changes (the interface change, and the additional types change), what about listing them as two points in the summary list.

Comment (matteofigus) [reply CHANGELOG.md:7]: Ok done
